### PR TITLE
fix more compile warnings

### PR DESCRIPTION
**Pull request by eder-matheus**

### CODE REVIEW

File: src/dpo/src/legalize_shift.h [hunk 1]
```diff
@@ -81,8 +81,6 @@ class ShiftLegalizer
   void merge(Clump* r);
   bool violated(Clump* r, Clump*& l, int& dist);
 
-  ShiftLegalizerParams& params_;
```
Comment (maliberty) [inline src/dpo/src/legalize_shift.h:84]: Should ShiftLegalizerParams be removed?  Similarly with other param classes

**Review by maliberty — COMMENTED** [review]
What compiler is generating these warnings?

**Comment by maliberty** [comment]
Please format as dpo was already formatted.

**Comment by eder-matheus** [comment]
> What compiler is generating these warnings?

I have the warnings on Apple clang version 13.1.6 (clang-1316.0.21.2.5)

File: src/dpo/src/legalize_shift.h (hunk 1 above)
Comment (eder-matheus) [reply src/dpo/src/legalize_shift.h:84]: Here are the warns I got for these params class members:

`OpenROAD/src/dpo/src/legalize_shift.h:84:25: warning: private field 'params_' is not used [-Wunused-private-field]
  ShiftLegalizerParams& params_;`

`OpenROAD/src/dpo/src/detailed_reorder.h:70:18: warning: private field 'rt_' is not used [-Wunused-private-field]
  RoutingParams* rt_;`

`OpenROAD/src/dpo/src/detailed_displacement.h:86:18: warning: private field 'rt_' is not used [-Wunused-private-field]
  RoutingParams* rt_;`

`OpenROAD/src/dpo/src/detailed_hpwl.h:97:18: warning: private field 'rt_' is not used [-Wunused-private-field]
  RoutingParams* rt_;`

`OpenROAD/src/dpo/src/detailed_abu.h:110:18: warning: private field 'rt_' is not used [-Wunused-private-field]
  RoutingParams* rt_;`

```
OpenROAD/src/dpo/src/detailed_orient.h:90:18: warning: private field 'rt_' is not used [-Wunused-private-field]
  RoutingParams* rt_;
```

It seems to me that the RoutingParams were copied and pasted from other classes.

Comment (maliberty) [reply src/dpo/src/legalize_shift.h:84]: Should ShiftLegalizerParams be removed?

Comment (eder-matheus) [reply src/dpo/src/legalize_shift.h:84]: Yes, it should, and I removed it in my last commit. I didn't get the question before, sorry.

**Review by maliberty — APPROVED** [review]

**Pull request merged**